src/lib.rs

```diff
@@ -177,9 +177,11 @@ fn rmain(config: &mut Config) -> Result<()> {
 
     for run in build.runs.iter() {
         config.status("Running", &format!("`{}`", run.join(" ")));
+        let (runtime_args, binary_args) = utils::split_args(run);
```

Comment:
I believe the way we'll want to handle this is do some option parsing *before* the `--` argument. Ideally this would use a full-fledged option parser, but we'll want to recognize `--runtime-args` as an argument to `cargo wasi run`, but not forward the argument to `cargo run`.

Put another way,` --runtime-args` is a flag to `cargo wasi run`, which means it needs to go before the `--`.

Reply:
> Ideally this would use a full-fledged option parser

which one would you suggest to use here?

Reply:
I don't have one in mind per se, I haven't looked into doing this yet.

Reply:
I just pushed an example using `structopt`. What do you think about it? Is it viable to continue this way?